### PR TITLE
Fixing list tile size in the presence of a large subtitle

### PR DESCRIPTION
**Pull request by gspencergoog**
Take into account the subtitle height when adjusting the height of the widget.  Added a test.

### CODE REVIEW

File: packages/flutter/test/material/list_tile_test.dart [hunk 1]
```diff
@@ -111,9 +112,12 @@ void main() {
     }
 
     void testVerticalGeometry(double expectedHeight) {
-      expect(tester.getSize(find.byType(ListTile)), new Size(800.0, expectedHeight));
-      if (hasSubtitle)
+      final Rect tileRect = tester.getRect(find.byType(ListTile));
```
Comment (HansMuller) [inline packages/flutter/test/material/list_tile_test.dart]: We should also verify that the top of the title is > 0. Once the MD team has signed of on a min vertical padding value (or some algorithm) we could refine testVerticalGeometry() further

File: packages/flutter/test/material/list_tile_test.dart [hunk 2]
```diff
@@ -111,9 +112,12 @@ void main() {
...
+      expect(tileRect.size, new Size(800.0, expectedHeight));
+      if (hasSubtitle) {
         expect(top('subtitle'), greaterThanOrEqualTo(bottom('title')));
+        expect(bottom('subtitle'), lessThanOrEqualTo(tileRect.bottom));
```
Comment (HansMuller) [inline packages/flutter/test/material/list_tile_test.dart]: I think this should be `lessThan` for now (because min vertical padding is > 0)

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/material/list_tile_test.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/test/material/list_tile_test.dart]: Done.  Also verified that the top centers the title in the widget.

File: packages/flutter/test/material/list_tile_test.dart (hunk 2 above)
Comment (gspencergoog) [reply packages/flutter/test/material/list_tile_test.dart]: Done

**Pull request merged**